### PR TITLE
An Overview of IoT Technology

### LINKED ISSUE #1825: Designing IoT Solutions
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Proposed title of article
Designing IoT Solutions
### Introduction paragraph (2-3 paragraphs):
The emergence of disruptive technologies such as Big data, AI, and Internet of Things (IoT) has brought a rapid growth in the Fourth Industrial revolution. The industrial revolution being data driven, is the primary reason for the rise in development of IoT applications and solutions.

IoT solutions and applications leverage on network of physical objects -''things'' that are embedded with sensors, software and data for the purposes of connecting and exchanging data over a network or the internet. This article will give readers an overview of what is takes to design IoT solutions.   
### Key takeaways: 
1. What is IoT
2. IoT architecture
3. IoT protocols
4. IoT development platforms and use cases
### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by OMONDIELLY**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1825

### CODE REVIEW

**Comment by OMONDIELLY** [comment]
hello @collins-a , I have made changes that are required.
Can I revise the title of the Article or help me change it to something like "Overview of IoT Technology"

**Review by collins-a — CHANGES_REQUESTED** [review]
Hey @OMONDIELLY, this is a pretty good first article. I especially enjoyed how you simplified your content. However, there are a few changes to be made before we get to the final review. Some of the sections came off as a bit ambiguous or lightweight. Take note of the review comments. Just some additions and we'll be good to go.

A couple more things:

1. Your hero image. Is it from Unsplash? It's also advisable to use an image that's as close as possible to 600x400 pixels. The current one well exceeds those dimesnions.
2. As a new contributor, I encourage you to add a brief Author Bio (index.md file) and Headshot image (avatar file). Include these files in the authors folder within the GitHub repo via your forked repo similar to the article files so as to be credited for your hard work.
3. Welcome to EngEd and don't hesitate to check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) pages to make your EngEd experience much easier.

File: articles/designing-iot-solutions/index.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
++ Handling communication protocols.
++ Ensuring security and identity of devices and applications.
++ Harmonizing data from the connected devices.
++ Providing multiple endpoints for various types of processing.
++ Providing industry communication protocols i.e. AMQP
+
+
+IoT hub is an example of development platforms for IoT solutions. Because IoT projects are complex, IoT hubs exist to fasten the development and delivery of solutions.
+
+Cloud platforms such as Azure host IoT hub where that acts as middleware for IoT solutions. Azure IoT hub provides reliable and secure communications between IoT devices and IoT applications that run on the cloud.
```
[the remark this answers is not in the document]
Comment (OMONDIELLY) [reply articles/designing-iot-solutions/index.md:165]: Sure , let me make the edits
Though I have Author page.

[the remark this answers is not in the document]
Comment (collins-a) [reply articles/designing-iot-solutions/index.md:164]: Looking through the submitted files, there is neither an author bio nor an avatar image. Do be sure to upload them

**Comment by collins-a** [comment]
Hey @OMONDIELLY, just checking in to make sure you're working on the changes I requested previously.

I've updated the article title. 

And a reminder to add a brief Author Bio (index.md file) and Headshot image (avatar file). Include these files in the authors folder within the GitHub repo via your forked repo, similar to the article files.

Let me know as soon as you're done, or ping me on Slack if you require assistance.

**Comment by OMONDIELLY** [comment]
Hi... actually this is my third article, so am not sure why my author folder is not showing.
I'll update it though.

**Comment by collins-a** [comment]
But do be keen with the changes you make on the article. eg. there are a couple of commits that have affected the flow of the article quite a bit. I've copied and pasted an example of a paragraph.

"What is IoT
<<<<<<< HEAD <<<<<<< HEAD Internet of Things (IoT) is a network of physical objects or devices – “things” – that are embedded with sensors, software, and other technologies for the purpose exchanging data among the devices on the internet to help organizations solve problems. For example, IoT solutions provide flexibility of workforce as employees can work remotely with connected devices, thus organizations incur little operational costs that relate to employee management. "

See the "<<<<<<< HEAD <<<<<<< HEAD" for instance. I'll rectify the ones I come across, but I suggest that you go through the article and get rid of duplicated sentences, paragraphs, and any elements that aren't meant to be present. Then ping me when done so that I may complete my review

**Comment by OMONDIELLY** [comment]
> But do be keen with the changes you make on the article. eg. there are a couple of commits that have affected the flow of the article quite a bit. I've copied and pasted an example of a paragraph.
> 
> "What is IoT
> <<<<<<< HEAD <<<<<<< HEAD Internet of Things (IoT) is a network of physical objects or devices – “things” – that are embedded with sensors, software, and other technologies for the purpose exchanging data among the devices on the internet to help organizations solve problems. For example, IoT solutions provide flexibility of workforce as employees can work remotely with connected devices, thus organizations incur little operational costs that relate to employee management. "
> 
> See the "<<<<<<< HEAD <<<<<<< HEAD" for instance. I'll rectify the ones I come across, but I suggest that you go through the article and get rid of duplicated sentences, paragraphs, and any elements that aren't meant to be present. Then ping me when done so that I may complete my review

Sure.. Am working on it, I think there was a problem on my end I  ended up pushing the same changes twice, sorry for that.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY I've just confirmed that it's true that you have an author bio. No need to update it. Must be an error with GitHub that's categorizing you as a first-time contributor.

So I should not make any changes?

**Comment by collins-a** [comment]
With regards to the author bio, unless advised otherwise, don't make any changes. Since it already exists

**Comment by OMONDIELLY** [comment]
> > But do be keen with the changes you make on the article. eg. there are a couple of commits that have affected the flow of the article quite a bit. I've copied and pasted an example of a paragraph.
> > "What is IoT
> > <<<<<<< HEAD <<<<<<< HEAD Internet of Things (IoT) is a network of physical objects or devices – “things” – that are embedded with sensors, software, and other technologies for the purpose exchanging data among the devices on the internet to help organizations solve problems. For example, IoT solutions provide flexibility of workforce as employees can work remotely with connected devices, thus organizations incur little operational costs that relate to employee management. "
> > See the "<<<<<<< HEAD <<<<<<< HEAD" for instance. I'll rectify the ones I come across, but I suggest that you go through the article and get rid of duplicated sentences, paragraphs, and any elements that aren't meant to be present. Then ping me when done so that I may complete my review
> 
> Sure.. Am working on it, I think there was a problem on my end I ended up pushing the same changes twice, sorry for that.

done making changes

**Review by collins-a — COMMENTED** [review]
@OMONDIELLY good job, the article is more polished now. However, I'll remind you to always do a Grammarly check before submission. I've made the edits for you this time. Also, you missed out on one change: AMQP. In comparison to MQTT, the section is still very lightweight. I'll still ask you to mention a bit more in terms of its usage cases, security, and/or other attributes of your choosing. Make it feel like it's more of a comparison with MQTT.

After that, we'll be ready for the final review.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY good job, the article is more polished now. However, I'll remind you to always do a Grammarly check before submission. I've made the edits for you this time. Also, you missed out on one change: AMQP. In comparison to MQTT, the section is still very lightweight. I'll still ask you to mention a bit more in terms of its usage cases, security, and/or other attributes of your choosing. Make it feel like it's more of a comparison with MQTT.
> 
> After that, we'll be ready for the final review


Am working on the changes

**Comment by OMONDIELLY** [comment]
Done with the changes

**Comment by collins-a** [comment]
@OMONDIELLY I can see you have deleted your index.md file. Ping me when it's available and done with the changes.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY I can see you have deleted your index.md file. Ping me when it's available and done with the changes.


Uploaded. Sorry about that , I had problems with my machine.

**Comment by OMONDIELLY** [comment]
I have made the changes requested you requested yesterday in the last commit . Kindly review.

**Review by collins-a — APPROVED** [review]
@OMONDIELLY I'm satisfied with the changes. Again, always remember to carry out Grammarly checks on your articles before submission and after making any changes.

**Comment by OMONDIELLY** [comment]
> @OMONDIELLY I'm satisfied with the changes. Again, always remember to carry out Grammarly checks on your articles before submission and after making any changes.

Sure, I'll be doing so

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**